frontend/src/components/pages/rp-connect/secrets/Secrets.Update.tsx

```diff
@@ -0,0 +1,134 @@
+import {Scope, UpdateSecretRequest} from '../../../../protogen/redpanda/api/dataplane/v1alpha2/secret_pb';
+
+const {ToastContainer, toast} = createStandaloneToast();
+
+const returnSecretTab = '/connect-clusters?defaultTab=redpanda-connect-secret'
+
+@observer
+class RpConnectSecretUpdate extends PageComponent<{ secretId: string }> {
+    @observable secret = '';
+    @observable isUpdating = false;
+
+    constructor(p: any) {
+        super(p);
+        makeObservable(this, undefined, {autoBind: true});
+    }
+
+    initPage(p: PageInitHelper) {
+        p.title = 'Update Secret';
+        p.addBreadcrumb('Redpanda Connect Secret Manager', '/rp-connect/secret/update');
+        p.addBreadcrumb('Update Secret', '');
+
+        this.refreshData(true);
+        appGlobal.onRefresh = () => this.refreshData(true);
+    }
+
+    refreshData(_force: boolean) {
+        rpcnSecretManagerApi.refreshSecrets(_force);
+    }
+
+    cancel() {
+        this.secret = '';
+        appGlobal.history.push(returnSecretTab);
+    }
+
+    async updateSecret() {
+        this.isUpdating = true;
+
+        //create function given string return base64 encoded Uint8Array
+        function base64Encode(str: string): Uint8Array {
+            const encodedString = btoa(str);
+            const charList = encodedString.split('').map(char => char.charCodeAt(0));
+            return new Uint8Array(charList);
+        }
+
+
+        rpcnSecretManagerApi.update(this.props.secretId, new UpdateSecretRequest({
+            id: this.props.secretId,
+            secretData: base64Encode(this.secret),
+            scopes: [Scope.REDPANDA_CONNECT]
+        }))
+            .then(async () => {
+                toast({
+                    status: 'success', duration: 4000, isClosable: false,
+                    title: 'Secret updated'
+                });
+                await pipelinesApi.refreshPipelines(true);
+                appGlobal.history.push(returnSecretTab);
+            })
+            .catch(err => {
+                toast({
+                    status: 'error', duration: null, isClosable: true,
+                    title: 'Failed to update secret',
+                    description: formatPipelineError(err),
+                })
+            })
+            .finally(() => {
+                this.isUpdating = false;
+            });
+    }
+
+    render() {
+        if (!rpcnSecretManagerApi.secrets) return DefaultSkeleton;
+
+        const isSecretEmpty = this.secret.trim().length == 0;
+
+        return (
+            <PageContent>
+                <ToastContainer/>
+                <Flex flexDirection="column" gap={5}>
+                    <FormField label="Secret name">
+                        <Flex alignItems="center" gap="2">
+                            <Input
+                                placeholder="Enter a secret name..."
+                                data-testid="secretId"
+                                pattern="[a-zA-Z0-9_\-]+"
```

Comment:
Maybe we can reuse this regex for secrets, actually I see this pattern is different
`^[A-Z][A-Z0-9_]*$` vs `[a-zA-Z0-9_\-]+`, which one should we use?

Reply:
I updated this based on proto